compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1969,10 +1969,24 @@ impl<'a: 'ast, 'ast, 'tcx> LateResolutionVisitor<'a, '_, 'ast, 'tcx> {
             && let ast::ExprKind::Path(None, _) = lhs.kind
         {
             if !ident_span.from_expansion() {
+                let (span, text) = match self.r.tcx.sess.source_map().span_to_snippet(ident_span) {
+                    Ok(var_name) if var_name.starts_with("let") => {
+                        // a special case for #117894
+                        let text = var_name.strip_prefix("let").and_then(|rest| {
+                            match rest.starts_with("_") {
+                                true => rest.strip_prefix("_").map(String::from),
+                                false => Some(rest.to_string())
+                            }
+                        }).unwrap_or(var_name);
+                        (ident_span, format!("let {}", text))
+                    },
```

Comment:
great, but here should use `Ok(ref name) if let Some(name) = name.strip_prefix("let") ...`, otherwise raise the following error:
```txt
error[E0505]: cannot move out of value because it is borrowed
    --> compiler/rustc_resolve/src/late/diagnostics.rs:1973:24
     |
1973 |                     Ok(name) if let Some(name) = name.strip_prefix("let") => {
     |                        ^^^^
     |                        |
     |                        move out of value occurs here
     |                        borrow of value occurs here
1974 |                         // a special case for #117894
1975 |                         let name = name.strip_prefix("_").unwrap_or(name);
     |                                    ---- borrow later used here
```